File: samples/15_three-layer-native/iosApp/Sources/HomeViewModel.swift
final class HomeViewModel {
    private let useCase: GetTasksUseCase

    init(useCase: GetTasksUseCase = GetTasksUseCase(repository: TaskRepository(dataSource: TaskDataSource()))) {
        self.useCase = useCase
    }

    func screenState(platform: String) -> String {
        useCase.loadTaskSummary(platform: platform)
    }
}

File: samples/kmp-native-ui/sharedLogic/src/commonMain/kotlin/com/example/kmpsamples/kmpnativeui/sharedlogic/HomeViewModel.kt
package com.example.kmpsamples.kmpnativeui.sharedlogic

class HomeViewModel(
    private val useCase: GetTasksUseCase = GetTasksUseCase(TaskRepository(TaskDataSource()))
) {
    fun screenState(platform: String): String = useCase.loadTaskSummary(platform)
}
